mmo.cc

```diff
@@ -704,7 +704,14 @@ mmo_read_CObjTrack(mmo_data_t* data)
 
     if (mmo_version >= 0x16) {
       // XXX ARB was u8 = gbfgetc(fin); but actually a string
-      QString text = mmo_readstr();
+      // Don't construct a QString we aren't going to use.
+      // avoid clazy-unused-non-trivial-variable
+#ifdef MMO_DBG
+      QString text =
```

Comment:
Can these be
Qstring junk = ...
Q_UNUSED(foo)
?

Reply:
This is a common pattern where the only usage is in a debug statement which imbeds another conditional test.    I don't particularly like cutting up the code with the preprocessor directives but I don't see a better alternative.  I think somewhere else we use preprocessor directives to split the non-debug and debug cases where the call to the read function appears in both, i.e.
```
#ifdef DEBUG
  QString junk = read()
  std::cout << junk << std:endl;
#else 
  (void) read();
#endif 
```

1. QString junk =, where junk is unused, will be flagged by clazy as [unused-non-trivial-variable](https://github.com/KDE/clazy/blob/master/docs/checks/README-unused-non-trivial-variable.md).
2. Q_UNUSED(foo) is an alternative to (void) foo, but only for the non-debug case.  Casting to void is a common solution independent of Qt.  Have a glance at [Q_UNUSED](https://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/global/qglobal.h?h=5.12.7#n117)